packages/harden/README.md

```diff
@@ -0,0 +1,102 @@
+# harden
+
+Hardened modules are modules that make their exports resist tampering by other
+modules that import them, making them less suceptible to supply chain attack.
+In [HardenedJS](https://hardenedjs.org), the global `harden` function
+transitively freezes an object and all of the objects that are reachable by
+walking up chains of properties and prototypes.
+All the primordials like `Array.prototype` and `Object` are frozen in
+this environment, which gives your module a place to stand toward its own
+defense.
+Then, with [LavaMoat](https://github.com/lavamoat/lavamoat), each package
+is credibly isolated and only receives the subset of globals and host modules
+it needs to function.
+That is, we can enforce [Principle of Least
+Authority](https://en.wikipedia.org/wiki/Principle_of_least_privilege).
+But, that leaves the module to use `harden` to freeze all its exports and
+anything it returns that might be shared by other packages that use it.
+
+In order to provide type information about the global `harden` in lockded-down
+HardenedJS, and also to make it possible for hardened modules to be used
+outside HardenedJS, the `@endo/harden` package exports a `harden` function that
+can be used either way.
+
+```js
+import { harden } from '@endo/harden';
+
+export const myFunction = () => {};
+harden(myFunction);
+
+export const MyConstructor = function MyConstructor() {};
+harden(MyConstructor.prototype);
```

Comment:
I may be off-base, but this line seems a little misleading, IIUC the latest `harden` design.  Under HardenedJS, isn't it supposed to at the very least recurse over own properties (i.e. `Reflect.ownKeys(obj)`), but not prototypes (i.e. `Reflect.getPrototypeOf(obj)`)?

With that in mind, `harden(MyConstructor.prototype)` seems completely redundant, since `.prototype` is an own property of `MyConstructor`, since it will be hardened upon calling `harden(MyConstructor)`.

Reading further, the proposed pre-lockdown `harden` is a no-op.  If it's never recursive (not even under HardenedJS), it just seems like an attractive nuisance.

Reply:
I think you are right that this is not a good example. I may need to appeal to @mhofman for a good example of a case where a defensively hardened module accidentally leaves something mutable exposed until the first instance is hardened. I’m likely confused and it’s possible that such a case is only possible with a speculatively different combination of alternative harden behaviors.